warehouse/slave_test.go

```diff
@@ -43,3 +61,195 @@ func TestPickupStagingFileBucket(t *testing.T) {
+		},
+		{
+			name: "Unknown provider",
+			conf: map[string]any{
+				"bucketProvider": "UNKNOWN",
+			},
+			wantError: errors.New("creating uploader: service provider not supported: UNKNOWN"),
+		},
+		{
+			name: "Invalid endpoint",
+			conf: map[string]any{
+				"endPoint": "http://localhost:1234",
+			},
+			wantError:         errors.New("uploading load file to object storage: uploading load file: Endpoint url cannot have fully qualified paths."),
+			additionalWriters: 9,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			minioResource, err := destination.SetupMINIO(pool, t)
+			require.NoError(t, err)
+
+			conf := map[string]any{
+				"bucketName":       minioResource.BucketName,
+				"accessKeyID":      minioResource.AccessKey,
+				"accessKey":        minioResource.AccessKey,
+				"secretAccessKey":  minioResource.SecretKey,
+				"endPoint":         minioResource.Endpoint,
+				"forcePathStyle":   true,
+				"s3ForcePathStyle": true,
+				"disableSSL":       true,
+				"region":           minioResource.SiteRegion,
+				"enableSSE":        false,
+				"bucketProvider":   provider,
+			}
+
+			for k, v := range tc.conf {
+				conf[k] = v
+			}
+
+			f, err := os.CreateTemp(t.TempDir(), "load.dump")
+			require.NoError(t, err)
+			t.Cleanup(func() { require.NoError(t, os.Remove(f.Name())) })
+
+			m := &mockLoadFileWriter{
+				file: f,
+			}
+
+			writerMap := map[string]encoding.LoadFileWriter{
+				"test": m,
+			}
+			for i := 0; i < tc.additionalWriters; i++ {
+				writerMap[fmt.Sprintf("test-%d", i)] = m
+			}
+
+			store := memstats.New()
+			stagingFileID := int64(1001)
+
+			destType := destType
+			if tc.destType != "" {
+				destType = tc.destType
+			}
+
+			job := Payload{
+				StagingFileID:            stagingFileID,
+				DestinationConfig:        conf,
+				UseRudderStorage:         false,
+				StagingDestinationConfig: conf,
+				StagingUseRudderStorage:  false,
+				WorkspaceID:              workspaceID,
+				DestinationID:            destinationID,
+				DestinationName:          destinationName,
+				SourceID:                 sourceID,
+				SourceName:               sourceName,
+				DestinationType:          destType,
+				LoadFilePrefix:           prefix,
+				UniqueLoadGenID:          uuid.New().String(),
+				DestinationNamespace:     namespace,
+			}
+			c := config.New()
+			c.Set("Warehouse.numLoadFileUploadWorkers", worker)
+			c.Set("Warehouse.slaveUploadTimeout", "5m")
+			c.Set("WAREHOUSE_BUCKET_LOAD_OBJECTS_FOLDER_NAME", loadObjectFolder)
+
+			jr := newJobRun(job, c, logger.NOP, store)
+			jr.since = func(t time.Time) time.Duration {
+				return time.Second
+			}
+			jr.outputFileWritersMap = writerMap
+
+			ctx := ctx
+			if tc.ctx != nil {
+				ctx = tc.ctx
+			}
+
+			loadFIle, err := jr.uploadLoadFiles(ctx)
+			if tc.wantError != nil {
+				require.EqualError(t, err, tc.wantError.Error())
+				return
+			}
```

Comment:
Could be simplified but you need to change `wantError` to `string`:

```suggestion
			if tc.wantError != "" {
				require.EqualError(t, err, tc.wantError)
				return
			}
```